### PR TITLE
fix: do not initialize distro in unsupported NodeJS versions

### LINKED ISSUE #166: Replace remove web services in integration/component tests with local http server
For example: https://github.com/postmanlabs/httpbin

### LINKED ISSUE #167: FileSpanExporter should not close the underpinning file descriptor
The fact that FileSpanExporter [closes the underpinning file descriptor](https://github.com/lumigo-io/opentelemetry-js-distro/blob/11f5dfc0af9372ff1455e10adf0ed709932b9acf/src/exporters/FileSpanExporter.ts#L92) causes issues with dockerized applications when the file it is writing to is `/dev/std{out|err}`.

### LINKED ISSUE #169: Build into startup checks for the Node.js version
When adding the OpenTelemetry JS distro on an unsupported Node.js version (e.g., Node.js 12), the distro causes the process to fail with the `fs/promises` package not being found. We need to check the version of node and bootstrap the distro in a safer manner on unsupported Node.js versions.

### PR DESCRIPTION
**Pull request by mmanciop**
Batch for fixes for multiple issues.

* Rework instrumentation test harness to be reliable, much smaller and written in TypeScript
* Unify component and integration tests into instrumentation tests
* Speed up and parallelize the build process
* Do not activate the distro is the Node.js version is not supported #169
* Rework FileSpanExporter not to close non-regular files #167
* Use local HTTP server for integration tests #166
* Add support for Node.js 20
* Update OpenTelemetry baseline to SDK v1.9.0, contrib 0.38.0 (or highest otherwise available)

Fixes #166
Fixes #167
Fixes #169

### CODE REVIEW

**Comment by mmanciop** [comment]
Relevant integration tests pass on CircleCI.

File: package.json [hunk 1]
```diff
@@ -2,22 +2,18 @@
...
   "version": "1.19.1",
   "description": "Lumigo wrapper to trace distributed architecture",
-  "main": "dist/src/wrapper.js",
+  "main": "dist/src/distro.js",
   "private": false,
-  "types": "dist/src/wrapper.d.ts",
+  "types": "dist/src/bootstrap.d.ts",
   "scripts": {
-    "build": "rm -rf dist wrapper.tgz && tsc --build --force && tracer=\"$(npm pack)\" && mv \"$tracer\" wrapper.tgz",
+    "build": "rm -rf dist *.tgz && tsc --build --force && tracer=\"$(npm pack)\" && mv \"$tracer\" distro.tgz && cp distro.tgz wrapper.tgz",
```
Comment (mmanciop) [inline package.json:9]: This is a bit of silliness not to break integration tests, that unfortunately have hard-coded the `wrapper.tgz` reference

File: package.json [hunk 2]
```diff
@@ -26,17 +22,16 @@
     "@lumigo/node-core": "^1.13.0",
     "@opentelemetry/api": "1.4.0",
     "@opentelemetry/core": "1.9.0",
-    "@opentelemetry/exporter-trace-otlp-http": "0.35.1",
-    "@opentelemetry/instrumentation": "0.35.1",
-    "@opentelemetry/instrumentation-express": "0.32.0",
```
Comment (mmanciop) [inline package.json:31]: This dependency was not even used

File: package.json [hunk 3]
```diff
@@ -2,22 +2,18 @@
   "name": "@lumigo/opentelemetry",
   "version": "1.19.1",
   "description": "Lumigo wrapper to trace distributed architecture",
-  "main": "dist/src/wrapper.js",
+  "main": "dist/src/distro.js",
   "private": false,
-  "types": "dist/src/wrapper.d.ts",
+  "types": "dist/src/bootstrap.d.ts",
```
Comment (mmanciop) [inline package.json:7]: Delicious life-hack to keep TS compatibility for end-users, without loading OpenTelemetry dependencies in the bootstrap until we know the Node.js version is safe.

**Review by doriaviram — APPROVED** [review]

**Review by GuyMoses — APPROVED** [review]
looks good!

I'm afraid that changing wrapper -> distro will break IT.
If does break - you'll need to modify IT code

**Pull request merged**